pkg/miscellaneous/miscellaneous.go

```diff
@@ -535,41 +540,81 @@ func ReadConfigFileAsMap(path string) (map[string]interface{}, error) {
+func parseAdvertiseURLsConfig(configFile string) (*advertiseURLsConfig, error) {
+	var advURLsConfig advertiseURLsConfig
+	config, err := os.ReadFile(configFile)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read etcd config file at path: %s : %w", configFile, err)
+	}
+
+	if err := yaml.Unmarshal(config, &advURLsConfig); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal etcd config yaml file at path: %s : %w", configFile, err)
 	}
-	domaiName := fmt.Sprintf("%s.%s.%s", tokens[1], tokens[2], "svc")
-	return fmt.Sprintf("%s://%s.%s:%s", tokens[0], podName, domaiName, tokens[3]), nil
+	return &advURLsConfig, nil
 }
 
-// IsPeerURLTLSEnabled checks whether the peer address is TLS enabled or not.
-func IsPeerURLTLSEnabled() (bool, error) {
-	podName, err := GetEnvVarOrError("POD_NAME")
+// GetInitialAdvertisePeerURLs retrieves the initial advertise peer URLs for the etcd member.
+func GetInitialAdvertisePeerURLs(configFile string) ([]string, error) {
+	memberName, err := GetEnvVarOrError("POD_NAME")
 	if err != nil {
-		return false, err
+		return nil, err
 	}
 
-	configFile := GetConfigFilePath()
+	advURLsConfig, err := parseAdvertiseURLsConfig(configFile)
+	if err != nil {
+		return nil, err
+	}
+
+	peerURLs, ok := advURLsConfig.InitialAdvertisePeerURLs[memberName]
+	if !ok || len(peerURLs) == 0 {
+		return nil, fmt.Errorf("peer url not found for pod %s", memberName)
+	}
 
-	config, err := ReadConfigFileAsMap(configFile)
+	for _, peerURL := range peerURLs {
+		if _, err := url.Parse(peerURL); err != nil {
+			return nil, fmt.Errorf("peer url %s is not valid", peerURL)
+		}
+	}
+	return peerURLs, nil
+}
+
+// GetAdvertiseClientURLs retrieves the advertise client URLs for the etcd member.
+func GetAdvertiseClientURLs(configFile string) ([]string, error) {
+	memberName, err := GetEnvVarOrError("POD_NAME")
 	if err != nil {
-		return false, err
+		return nil, err
 	}
-	initAdPeerURL := config["initial-advertise-peer-urls"]
 
-	memberPeerURL, err := ParsePeerURL(initAdPeerURL.(string), podName)
+	advURLsConfig, err := parseAdvertiseURLsConfig(configFile)
 	if err != nil {
-		return false, err
+		return nil, err
+	}
+
+	clientURLs, ok := advURLsConfig.AdvertiseClientURLs[memberName]
+	if !ok || len(clientURLs) == 0 {
+		return nil, fmt.Errorf("client url not found for pod %s", memberName)
 	}
 
-	peerURL, err := url.Parse(memberPeerURL)
+	for _, clientURL := range clientURLs {
+		if _, err := url.Parse(clientURL); err != nil {
+			return nil, fmt.Errorf("client url %s is not valid", clientURL)
+		}
+	}
+	return clientURLs, nil
+}
+
+// IsPeerURLTLSEnabled checks whether the peer address is TLS enabled or not.
+func IsPeerURLTLSEnabled() (bool, error) {
+	configFile := GetConfigFilePath()
+	memberPeerURLs, err := GetInitialAdvertisePeerURLs(configFile)
+	if err != nil {
+		return false, err
+	}
+	peerURL, err := url.Parse(memberPeerURLs[0])
```

Comment:
Check whether any of the peer URLs are TLS-enabled, you can adjust the function to iterate over all provided URLs and return true as soon as it finds one with the HTTPS scheme

Reply:
To be more accurate, I'm now iterating over the urls and even if one of them is `http` I'm returning false. Since, for us to say peer urls are TLS enabled, every one of them should be `https`. I hope this is okay.